js/app.js

```diff
@@ -68,18 +68,22 @@ Event.on(document.body, 'ui:click', '.show-tutorial', function(evt){
 });
 // Documentation for modal dialogs: https://github.com/kylepaulsen/NanoModal
 
-Event.on(document.body, 'ui:click', '.do-run', startScript);
-Event.on(document.body, 'ui:click', '.do-stop', stopScript);
+Event.on('.do-run', 'ui:click', null, startScript);
+Event.on('.do-stop', 'ui:click', null, stopScript);
 
-function startScript() {
+function startScript(evt){
     // Do any necessary cleanup (e.g., clear event handlers).
-    stopScript();
-    runtime.resetCanvas();
+    stopScript(evt);
+    runtime.resetStage();
+    evt.target.blur();
+    runtime.getStage().focus();
     preload().whenLoaded(runScript);
 }
 
-function stopScript() {
+function stopScript(evt){
     runtime.stopEventLoop();
+    evt.target.blur();
+    runtime.getStage().focus()
```

Comment:
So when the user clicks stop, we want the mouse focus to be on the stage? I guess that seems predictable.